src/main/java/com/thoughtworks/gauge/execution/StepExecutionStage.java

```diff
@@ -85,10 +88,14 @@ public void setNextStage(ExecutionStage stage) {
             Class<?>[] parameterTypes = method.getParameterTypes();
             for (int i = 0; i < parameterTypes.length; i++) {
                 Class<?> parameterType = parameterTypes[i];
-                if (primitiveConverters.containsKey(parameterType)) {
-                    parameters[i] = primitiveConverters.get(parameterType).convert(args.get(i));
+                Parameter parameter = args.get(i);
+                if(isTable(parameter)){
```

Comment:
I've not done in this PR to keep simplicity but I think this if/else block should be extracted in a factory or something like that.

Tell me if you think it would be better.